### PR TITLE
Create 7.4.x formula

### PR DESCRIPTION
**Pull request by jondavidjohn**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-versions/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
- [x] Does your submission adhere to Versions "Acceptable Formulae" [guidelines](https://github.com/Homebrew/homebrew-versions/blob/master/README.md#acceptable-formulae)?

-----

Regarding #1421

### CODE REVIEW

File: vim74.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  url "https://github.com/vim/vim/archive/v7.4.2367.tar.gz"
+  sha256 "a9ae4031ccd73cc60e771e8bf9b3c8b7f10f63a67efce7f61cd694cd8d7cda5c"
+  head "https://github.com/vim/vim.git"
+
+  bottle do
+    sha256 "3abdbb854b7991d130804c4de81d6cf45cf5189acb90dbfa87ebb8207f9f5c20" => :sierra
+    sha256 "0281cba2aab18a0bcb5dce187aa652cbfcfc8b424f8683ca94a0252cc2240237" => :el_capitan
+    sha256 "8fe47d3288c426ac415e939366b706ac4579c6a6ab9a1a9f8785575f992c39a2" => :yosemite
+    sha256 "ace00ef6ad044ff2779c8f99fac6d5db3f91c2751f9dc7a9198ca911480a9cca" => :mavericks
+  end
```
Comment (MikeMcQuaid) [inline vim74.rb]: Please remove the bottle block.

File: vim74.rb [hunk 2]
```diff
@@ -0,0 +1,128 @@
+class Vim74 < Formula
+  desc "Vi \"workalike\" with many additional features"
+  homepage "http://www.vim.org/"
+  # *** Vim should be updated no more than once every 7 days ***
```
Comment (MikeMcQuaid) [inline vim74.rb]: Can remove this comment.

File: vim74.rb [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  url "https://github.com/vim/vim/archive/v7.4.2367.tar.gz"
+  sha256 "a9ae4031ccd73cc60e771e8bf9b3c8b7f10f63a67efce7f61cd694cd8d7cda5c"
+  head "https://github.com/vim/vim.git"
```
Comment (MikeMcQuaid) [inline vim74.rb]: Can remove this or point to 7.4 branch.

**Comment by jondavidjohn** [comment]
@MikeMcQuaid Should be ready to look at again, thanks!

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `vim74: 7.4.2367 (new formula)`. Please complete the issue template in future PRs where this format is detailed in the linked [Contributing guidelines](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md). Thanks!

**Pull request closed without merging**